tasks/eva_2205/src/main/groovy/eva2205/eva2205_undo_merges_from_multimaps.groovy

```diff
@@ -0,0 +1,67 @@
+import static uk.ac.ebi.eva.groovy.commons.EVADatabaseEnvironment.*
+import static org.springframework.data.mongodb.core.query.Criteria.where
+import groovy.cli.picocli.CliBuilder
+
+def cli = new CliBuilder()
+cli.propertiesFile(args:1, "Properties file to use for remediation", required: true)
+cli.assembly(args:1, "Assembly to remediate", required: true)
+def options = cli.parse(args)
+if (!options) {
+    cli.usage()
+    System.exit(1)
+}
+
+def dbEnv = createFromSpringContext(options.propertiesFile, Application.class,
+        ["parameters.assemblyAccession": options.assembly])
+def logger = LoggerFactory.getLogger(GenericApplication.class)
+
+def ssWithMapWtMergeOps = [svoeClass, dbsnpSvoeClass].collect{new RetryableBatchingCursor<>(
+        where("inactiveObjects.seq").is(options.assembly).and("eventType").is(EventType.UPDATED.toString()).and(
+                "reason").regex(".* merged into rs.*").and("inactiveObjects.mapWeight").exists(true),
+        dbEnv.mongoTemplate, it)}
+ssWithMapWtMergeOps.each {it.each{List<SubmittedVariantOperationEntity> ops ->
+    def (sveBulkUpdates, dbsnpSveBulkUpdates) = [sveClass, dbsnpSveClass].collect{
+        dbEnv.mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED, it)}
+    def opsGroupedByHash = ops.groupBy{it.inactiveObjects[0].hashedMessage}
+    def ssHashesThatShouldHaveMapWt = ops.collect{it.inactiveObjects[0].hashedMessage}.unique()
+    def (svesThatShouldGetMapWt, dbsnpSvesThatShouldGetMapWt) = [sveClass, dbsnpSveClass].collect{
+        dbEnv.mongoTemplate.find(query(where("_id").in(ssHashesThatShouldHaveMapWt).and("mapWeight").exists(false)), it)}
+    def updateTargetSves = {targetSvesToUpdate, bulkUpdateObj, opsClass ->
+        if (targetSvesToUpdate.size() > 0) {
+            def opsToInsert = new ArrayList<>()
+            targetSvesToUpdate.each {sve ->
+                def correspondingOp = opsGroupedByHash.get(sve.hashedMessage)[0]
+                def mapWeightedRSToAssign = correspondingOp.inactiveObjects[0].clusteredVariantAccession
+                def updatesToMake = new Update()
+                updatesToMake.set("mapWeight", correspondingOp.inactiveObjects[0].mapWeight)
+                updatesToMake.set("rs", mapWeightedRSToAssign)
+                bulkUpdateObj.updateOne(query(where("_id").is(sve.hashedMessage)), updatesToMake)
+                logger.info("SS with hash ${sve.hashedMessage} and accession ${sve.accession} will be assigned " +
+                        "map-weighted RS ${mapWeightedRSToAssign}")
+
+                def opToRecordRestoredMapWt = new SubmittedVariantOperationEntity()
+                opToRecordRestoredMapWt.fill(EventType.UPDATED, sve.accession, "Restore map-weighted RS ID.",
+                        Arrays.asList(new SubmittedVariantInactiveEntity(sve)))
+                opToRecordRestoredMapWt.setId("RESTORE_MAPWT_${sve.hashedMessage}")
+                opsToInsert.add(opToRecordRestoredMapWt)
```

Comment:
Should this undo process also have an associated CVOE?

Reply:
I had thought about this and I came to the following conclusion:

Resurrecting a map-weighted RS could possibly involve kicking out existing legitimate RS with the same locus (at least temporarily). Doing that seemed problematic and didn't seem to be worth the trouble it entails. Given that decision of not resurrecting these RS: since these RS were already merged and not in the "current" collection, recording a deprecation operation in CVOE for those RS might also be misleading.

Reply:
I am a bit worried about [this scenario](https://docs.google.com/spreadsheets/d/1kRKHR4zrq-nxH_Sg82TwXZbxdgjB3K-q4YMyJva8yMg/edit#rangeid=739110006) though...